<img src=".\images\Logo.png" width=150 align="left" /> <img src=".\images\Logo2.jpg" width=450 align="right" />


# <center><font color= #1e8449 > <b>CU34_Predicción de demanda de servicios</font></center>

<font color='steelblue'><b>Citizenlab Data Science Methodology > II - Data Processing Domain </font>
***
> # <font color='steelblue'> <b>05.- Data Collection</font>

<font color='steelblue'>Data Collection is the process to obtain and generate (if required) necessary data to model the problem.</font>

### 04. Estimar datos meteorológicos en secciones censales

* Estimar variables en secciones
* Eliminando variables que no se usarán en modelos




<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Settings" data-toc-modified-id="Settings-1"><font color="green">Settings</font></a></span></li><li><span><a href="#Data-Load" data-toc-modified-id="Data-Load-2"><font color="green">Data Load</font></a></span></li><li><span><a href="#ETL-Processes" data-toc-modified-id="ETL-Processes-3">ETL Processes</a></span><ul class="toc-item"><li><span><a href="#Import-data-from:-CSV,-Excel,-Tab,-JSON,-SQL,-and-Parquet-files" data-toc-modified-id="Import-data-from:-CSV,-Excel,-Tab,-JSON,-SQL,-and-Parquet-files-3.1">Import data from: CSV, Excel, Tab, JSON, SQL, and Parquet files</a></span></li></ul></li><li><span><a href="#Synthetic-Data-Generation" data-toc-modified-id="Synthetic-Data-Generation-4">Synthetic Data Generation</a></span></li><li><span><a href="#Fake-Data-Generation" data-toc-modified-id="Fake-Data-Generation-5">Fake Data Generation</a></span></li><li><span><a href="#Open-Data" data-toc-modified-id="Open-Data-6">Open Data</a></span></li><li><span><a href="#Data-Save" data-toc-modified-id="Data-Save-7"><font color="green">Data Save</font></a></span></li><li><span><a href="#Main-Conclusions" data-toc-modified-id="Main-Conclusions-8"><font color=" #2874a6 ">Main Conclusions</font></a></span></li><li><span><a href="#Main-Actions" data-toc-modified-id="Main-Actions-9"><font color=" #2874a6 ">Main Actions</font></a></span><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Acciones-done" data-toc-modified-id="Acciones-done-9.0.1"><font color=" #2874a6 ">Acciones done</font></a></span></li><li><span><a href="#Acctions-to-perform" data-toc-modified-id="Acctions-to-perform-9.0.2"><font color=" #2874a6 ">Acctions to perform</font></a></span></li></ul></li></ul></li></ul></div>

## <font color='green'>Settings</font>

### Encoding

Con la siguiente expresión se evitan problemas con el encoding al ejecutar el notebook. Es posible que deba ser eliminada o adaptada a la máquina en la que se ejecute el código.

In [1]:
Sys.setlocale(category = "LC_ALL", locale = "es_ES.UTF-8")

[1] "es_ES.UTF-8/es_ES.UTF-8/es_ES.UTF-8/C/es_ES.UTF-8/C"

### Packages to use

- {tcltk} para selección interactiva de archivos locales
- {sf} para trabajar con georeferenciación
- {gstat} para cálculos geoestadísticos
- {readr} para leer y escribir archivos csv
- {dplyr} para explorar datos
- {stringr} para manipulación de cadenas de caracteres
- {tidyr} para quitar perdidos
- {lubridate} para manipulación de fechas


In [2]:
library(readr)
library(dplyr)
library(sf)
library(gstat)
library(stringr)
library(tidyr)
library(lubridate)



Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union


Linking to GEOS 3.10.2, GDAL 3.4.2, PROJ 8.2.1; sf_use_s2() is TRUE


Attaching package: ‘lubridate’


The following objects are masked from ‘package:base’:

    date, intersect, setdiff, union




### Paths

In [3]:
iPath <- "Data/Input/"
oPath <- "Data/Output/"

## <font color='green'>Data Load</font>

If there are more than one input file, make as many sections as files to import.

1. Datos meteorológicos de estaciones

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [4]:
# file_data_01 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [5]:
iFile_01 <- "CU_34_05_03_aemet.csv"
file_data_01 <- paste0(iPath, iFile_01)

if(file.exists(file_data_01)){
    cat("Se leerán datos del archivo: ", file_data_01)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_34_05_03_aemet.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [6]:
data_01 <- read_csv(file_data_01)


Rows: 372 Columns: 22
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (9): indicativo, nombre, provincia, horatmin, horatmax, dir, horaracha...
dbl  (12): altitud, tmed, prec, tmin, tmax, velmedia, racha, sol, presMax, p...
date  (1): fecha

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [7]:
glimpse(data_01)

Rows: 372
Columns: 22
$ fecha       <date> 2022-01-01, 2022-01-02, 2022-01-03, 2022-01-04, 2022-01-0…
$ indicativo  <chr> "2462", "2462", "2462", "2462", "2462", "2462", "2462", "2…
$ nombre      <chr> "PUERTO DE NAVACERRADA", "PUERTO DE NAVACERRADA", "PUERTO …
$ provincia   <chr> "MADRID", "MADRID", "MADRID", "MADRID", "MADRID", "MADRID"…
$ altitud     <dbl> 1894, 1894, 1894, 1894, 1894, 1894, 1894, 1894, 1894, 1894…
$ tmed        <dbl> 10.6, 9.8, 7.8, 1.6, -3.0, -5.6, -4.0, -2.8, 0.0, 1.6, 1.0…
$ prec        <dbl> 0.0, 0.0, 0.0, 24.2, 13.9, 0.0, 0.2, 0.0, 6.6, 0.8, 0.0, 0…
$ tmin        <dbl> 7.4, 6.0, 4.2, -3.9, -4.2, -8.3, -6.6, -4.6, -3.0, 0.3, -0…
$ horatmin    <chr> "00:10", "20:40", "06:20", "23:59", "03:10", "06:40", "03:…
$ tmax        <dbl> 13.7, 13.5, 11.3, 7.0, -1.8, -2.8, -1.3, -0.9, 3.0, 3.0, 2…
$ horatmax    <chr> "Varias", "13:30", "Varias", "01:40", "Varias", "00:00", "…
$ dir         <chr> "22", "22", "27", "33", "32", "35", "36", "33", "31", "33"…
$ velmedia    <dbl

Muestra de datos:

In [8]:
slice_head(data_01, n = 5)

fecha      indicativo nombre                provincia altitud tmed prec tmin
1 2022-01-01 2462       PUERTO DE NAVACERRADA MADRID    1894    10.6  0.0  7.4
2 2022-01-02 2462       PUERTO DE NAVACERRADA MADRID    1894     9.8  0.0  6.0
3 2022-01-03 2462       PUERTO DE NAVACERRADA MADRID    1894     7.8  0.0  4.2
4 2022-01-04 2462       PUERTO DE NAVACERRADA MADRID    1894     1.6 24.2 -3.9
5 2022-01-05 2462       PUERTO DE NAVACERRADA MADRID    1894    -3.0 13.9 -4.2
  horatmin tmax ... velmedia racha horaracha sol presMax horaPresMax presMin
1 00:10    13.7 ... 3.9      12.2  22:30     8.1 823.3   11          821.8  
2 20:40    13.5 ... 2.5      11.1  00:00     6.0 823.5   Varias      821.5  
3 06:20    11.3 ... 3.1      15.3  22:50     7.6 821.5   00          814.7  
4 23:59     7.0 ... 7.8      23.6  18:30     3.3 814.7   00          803.6  
5 03:10    -1.8 ... 4.2      19.2  00:30     0.0 809.5   11          806.4  
  horaPresMin X         Y       
1 06          -4.010556 40.79306
2 Varias      -4.010556 40.79306
3 24          -4.010556 40.79306
4 17          -4.010556 40.79306
5 Varias      -4.010556 40.79306

2. Datos de contornos secciones

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [9]:
# file_data_02 <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [10]:
iFile_02 <- "CU_34_05_01_secciones_geo.json"
file_data_02 <- paste0(iPath, iFile_02)

if(file.exists(file_data_02)){
    cat("Se leerán datos del archivo: ", file_data_02)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leerán datos del archivo:  Data/Input/CU_34_05_01_secciones_geo.json

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [11]:
data_02 <- st_read(file_data_02)


Reading layer `CU_34_05_01_secciones_geo' from data source 
  `/Users/emilio.lcano/academico/gh_repos/__transferencia/citizenlab/CitizenLab-Research-and-Development/casos_urjc/notebooks/II_data_processing/34_servicios/Data/Input/CU_34_05_01_secciones_geo.json' 
  using driver `GeoJSON'
Simple feature collection with 4432 features and 16 fields
Geometry type: MULTIPOLYGON
Dimension:     XY
Bounding box:  xmin: -4.579006 ymin: 39.8848 xmax: -3.052983 ymax: 41.16584
Geodetic CRS:  WGS 84


Estructura de  los datos:

In [12]:
glimpse(data_02)

Rows: 4,432
Columns: 17
$ CUSEC    <chr> "2800101001", "2800201001", "2800201002", "2800301001", "2800…
$ CUMUN    <chr> "28001", "28002", "28002", "28003", "28004", "28004", "28004"…
$ CSEC     <chr> "001", "001", "002", "001", "001", "002", "003", "004", "001"…
$ CDIS     <chr> "01", "01", "01", "01", "01", "01", "01", "01", "01", "01", "…
$ CMUN     <chr> "001", "002", "002", "003", "004", "004", "004", "004", "005"…
$ CPRO     <chr> "28", "28", "28", "28", "28", "28", "28", "28", "28", "28", "…
$ CCA      <chr> "13", "13", "13", "13", "13", "13", "13", "13", "13", "13", "…
$ CUDIS    <chr> "2800101", "2800201", "2800201", "2800301", "2800401", "28004…
$ CLAU2    <chr> "28001", "28002", "28002", "28003", "28004", "28004", "28004"…
$ NPRO     <chr> "Madrid", "Madrid", "Madrid", "Madrid", "Madrid", "Madrid", "…
$ NCA      <chr> "Comunidad de Madrid", "Comunidad de Madrid", "Comunidad de M…
$ CNUT0    <chr> "ES", "ES", "ES", "ES", "ES", "ES", "ES", "ES", "ES", "ES", "…
$ CNUT1    <chr>

Muestra de datos:

In [13]:
data_02 |> tibble() |> slice_head(n = 5)

CUSEC      CUMUN CSEC CDIS CMUN CPRO CCA CUDIS   CLAU2 NPRO  
1 2800101001 28001 001  01   001  28   13  2800101 28001 Madrid
2 2800201001 28002 001  01   002  28   13  2800201 28002 Madrid
3 2800201002 28002 002  01   002  28   13  2800201 28002 Madrid
4 2800301001 28003 001  01   003  28   13  2800301 28003 Madrid
5 2800401001 28004 001  01   004  28   13  2800401 28004 Madrid
  NCA                 CNUT0 CNUT1 CNUT2 CNUT3 NMUN             
1 Comunidad de Madrid ES    3     0     0     Acebeda, La      
2 Comunidad de Madrid ES    3     0     0     Ajalvir          
3 Comunidad de Madrid ES    3     0     0     Ajalvir          
4 Comunidad de Madrid ES    3     0     0     Alameda del Valle
5 Comunidad de Madrid ES    3     0     0     Álamo, El        
  geometry                      
1 MULTIPOLYGON (((-3.64316 41...
2 MULTIPOLYGON (((-3.469465 4...
3 MULTIPOLYGON (((-3.503032 4...
4 MULTIPOLYGON (((-3.808662 4...
5 MULTIPOLYGON (((-3.979038 4...

## ETL Processes

### Import data from: CSV, Excel, Tab, JSON, SQL, and Parquet files

Se han importado en el apartado Data Load anterior:

* Datos diarios meteorológicos por estación
* Contornos de secciones censales

Incluir apartados si procede para: Extracción de datos (select, filter), Transformación de datos, (mutate, joins, ...). Si es necesario tratar datos perdidos, indicarlo también en NB 09.2

Si no aplica: Estos datos no requieren tareas de este tipo.

#### Data transformation

* Convertir zonas a centroide para poder interpolar

In [14]:
tdata_02 <- data_02 |> st_centroid() 

Warning message in st_centroid.sf(data_02):
“st_centroid assumes attributes are constant over geometries of x”


In [15]:
glimpse(tdata_02)

Rows: 4,432
Columns: 17
$ CUSEC    <chr> "2800101001", "2800201001", "2800201002", "2800301001", "2800…
$ CUMUN    <chr> "28001", "28002", "28002", "28003", "28004", "28004", "28004"…
$ CSEC     <chr> "001", "001", "002", "001", "001", "002", "003", "004", "001"…
$ CDIS     <chr> "01", "01", "01", "01", "01", "01", "01", "01", "01", "01", "…
$ CMUN     <chr> "001", "002", "002", "003", "004", "004", "004", "004", "005"…
$ CPRO     <chr> "28", "28", "28", "28", "28", "28", "28", "28", "28", "28", "…
$ CCA      <chr> "13", "13", "13", "13", "13", "13", "13", "13", "13", "13", "…
$ CUDIS    <chr> "2800101", "2800201", "2800201", "2800301", "2800401", "28004…
$ CLAU2    <chr> "28001", "28002", "28002", "28003", "28004", "28004", "28004"…
$ NPRO     <chr> "Madrid", "Madrid", "Madrid", "Madrid", "Madrid", "Madrid", "…
$ NCA      <chr> "Comunidad de Madrid", "Comunidad de Madrid", "Comunidad de M…
$ CNUT0    <chr> "ES", "ES", "ES", "ES", "ES", "ES", "ES", "ES", "ES", "ES", "…
$ CNUT1    <chr>

In [16]:
tdata_02 |> tibble() |> slice_head(n = 5)

CUSEC      CUMUN CSEC CDIS CMUN CPRO CCA CUDIS   CLAU2 NPRO  
1 2800101001 28001 001  01   001  28   13  2800101 28001 Madrid
2 2800201001 28002 001  01   002  28   13  2800201 28002 Madrid
3 2800201002 28002 002  01   002  28   13  2800201 28002 Madrid
4 2800301001 28003 001  01   003  28   13  2800301 28003 Madrid
5 2800401001 28004 001  01   004  28   13  2800401 28004 Madrid
  NCA                 CNUT0 CNUT1 CNUT2 CNUT3 NMUN             
1 Comunidad de Madrid ES    3     0     0     Acebeda, La      
2 Comunidad de Madrid ES    3     0     0     Ajalvir          
3 Comunidad de Madrid ES    3     0     0     Ajalvir          
4 Comunidad de Madrid ES    3     0     0     Alameda del Valle
5 Comunidad de Madrid ES    3     0     0     Álamo, El        
  geometry                  
1 POINT (-3.63571 41.09315) 
2 POINT (-3.475764 40.5202) 
3 POINT (-3.496712 40.53807)
4 POINT (-3.84994 40.92032) 
5 POINT (-3.985307 40.24362)

#### Data extract

* Seleccionar variables temperatura, precipitaciones y presión

In [17]:
edata_01 <- data_01 |>
   select(fecha, nombre, tmed, prec, velmedia, presMax, X, Y) 

#### Data transformation

* Convertir datos a objeto espacial para poder interpolar


In [18]:
tdata_01 <- edata_01 |>
    st_as_sf(coords = c("X", "Y"), crs = 4326)

In [19]:
glimpse(tdata_01)

Rows: 372
Columns: 7
$ fecha    <date> 2022-01-01, 2022-01-02, 2022-01-03, 2022-01-04, 2022-01-05, …
$ nombre   <chr> "PUERTO DE NAVACERRADA", "PUERTO DE NAVACERRADA", "PUERTO DE …
$ tmed     <dbl> 10.6, 9.8, 7.8, 1.6, -3.0, -5.6, -4.0, -2.8, 0.0, 1.6, 1.0, 2…
$ prec     <dbl> 0.0, 0.0, 0.0, 24.2, 13.9, 0.0, 0.2, 0.0, 6.6, 0.8, 0.0, 0.0,…
$ velmedia <dbl> 3.9, 2.5, 3.1, 7.8, 4.2, 5.6, 4.2, 5.6, 6.1, 6.1, 2.5, 1.7, 5…
$ presMax  <dbl> 823.3, 823.5, 821.5, 814.7, 809.5, 813.5, 819.3, 819.2, 815.1…
$ geometry <POINT [°]> POINT (-4.010556 40.79306), POINT (-4.010556 40.79306),…


In [20]:
tdata_01 |> tibble() |> slice_head(n = 5)

fecha      nombre                tmed prec velmedia presMax
1 2022-01-01 PUERTO DE NAVACERRADA 10.6  0.0 3.9      823.3  
2 2022-01-02 PUERTO DE NAVACERRADA  9.8  0.0 2.5      823.5  
3 2022-01-03 PUERTO DE NAVACERRADA  7.8  0.0 3.1      821.5  
4 2022-01-04 PUERTO DE NAVACERRADA  1.6 24.2 7.8      814.7  
5 2022-01-05 PUERTO DE NAVACERRADA -3.0 13.9 4.2      809.5  
  geometry                  
1 POINT (-4.010556 40.79306)
2 POINT (-4.010556 40.79306)
3 POINT (-4.010556 40.79306)
4 POINT (-4.010556 40.79306)
5 POINT (-4.010556 40.79306)

* Interpolar temperatura media, precipitación, velocidad media del viento y presión máxima en las coordenadas de cada sección censal.

NOTA: esta operación puede tardar varios minutos.

In [21]:
meteovars <- c("tmed", "prec", "velmedia", "presMax")
dias <- tdata_01$fecha |> unique()
resl <- list()

for (i in seq_along(dias)){
  f <- dias[i]
  meteod <- tdata_01 |> 
    filter(fecha == f) |> 
    select(all_of(meteovars)) 
  
  res <- data_02 |> select(CMUN, CDIS, CSEC) |> st_drop_geometry()
  for (j in seq_along(meteovars)){
    thisvar <- meteovars[j] 
    meteodc <- meteod |> 
      drop_na(all_of(thisvar))
    
    if(all(meteodc |> pull(thisvar) == 0)){
      res <- res |> 
        mutate(!!thisvar := 0)
    } else{
      
      fo <- as.formula(paste0(thisvar, " ~ 1"))
      v0 <- variogram(fo, meteodc)
      v.m <- suppressWarnings(fit.variogram(v0, 
        vgm(c("Exp", "Mat", "Sph", "Log", "Wav", "Bes", "Lin", "Leg"))))
      invisible(capture.output(suppressWarnings(krg <- krige(fo, 
                   locations = meteodc,
                   newdata = tdata_02,
                   model = v.m))))
      res <- res |> 
        mutate(fecha = f) |> 
        mutate(!!thisvar := krg$var1.pred)
    }
  }
    # res <- res |>
    #   mutate(ano = a,
    #          semana = s)
  resl[[i]] <- res
}

data <- bind_rows(resl)

In [22]:
glimpse(data)

Rows: 137,392
Columns: 8
$ CMUN     <chr> "001", "002", "002", "003", "004", "004", "004", "004", "005"…
$ CDIS     <chr> "01", "01", "01", "01", "01", "01", "01", "01", "01", "01", "…
$ CSEC     <chr> "001", "001", "002", "001", "001", "002", "003", "004", "001"…
$ fecha    <date> 2022-01-01, 2022-01-01, 2022-01-01, 2022-01-01, 2022-01-01, …
$ tmed     <dbl> 9.472965, 8.170448, 8.526402, 10.012282, 10.574368, 10.508450…
$ prec     <dbl> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0…
$ velmedia <dbl> 0.8229741, 0.9750191, 0.8874318, 2.3162929, 0.6970652, 0.6875…
$ presMax  <dbl> 901.6717, 970.1249, 967.2948, 849.5611, 964.2144, 965.4248, 9…


## Synthetic Data Generation

No aplica

## Fake Data Generation

No aplica

## Open Data

Los datos se obtuvieron de fuentes abiertas en la tarea anterior

## <font color='green'>Data Save</font>

Este proceso, puede copiarse y repetirse en aquellas partes del notebbok que necesiten guardar datos.
Recuerde cambiar la extensión añadida del fichero para diferenciarlas

<font color='tomato'> Identificamos los datos a guardar</font>

In [23]:
data_to_save <- data


<font color='tomato'>Estructura de nombre de archivos:</font>

* Código del caso de uso, por ejemplo "CU_04"
* Número del proceso que lo genera, por ejemplo "_05".
* Número de la tarea que lo genera, por ejemplo "_01"
* En caso de generarse varios ficheros en la misma tarea, llevarán _01 _02 ... después
* Nombre: identificativo de "properData", por ejemplo "_zonasgeo"
* Extensión del archivo

Ejemplo: "CU_04_05_01_01_zonasgeo.json, primer fichero que se genera en la tarea 01 del proceso 05 (Data Collection) para el caso de uso 04 (vacunas)

Importante mantener los guiones bajos antes de proceso, tarea, archivo y nombre

### Proceso 05

In [24]:
caso <- "CU_34"
proceso <- '_05'
tarea <- "_04"
archivo <- ""
proper <- "_meteo_secciones"
extension <- ".csv"

<font color='tomato'><b> OPCION A:</b> Uso del paquete "tcltk" para mayor comodidad</font>

* Buscar carpeta, escribir nombre de archivo SIN extensión (se especifica en el código)
* Especificar sufijo2 si es necesario
* Cambiar datos por datos_xx si es necesario

In [25]:
# file_save <- paste0(caso, proceso, tarea, tcltk::tkgetSaveFile(), proper, extension) 
# path_out <- paste0(oPath, file_save)
# write_csv(data_to_save, path_out)

# cat('File saved as: ')
# path_out

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

- Los ficheros de salida del proceso van siempre a Data/Output/.  

In [26]:
file_save <- paste0(caso, proceso, tarea, archivo, proper, extension) 
path_out <- paste0(oPath, file_save)
write_csv(data_to_save, path_out)

cat('File saved as: ')
path_out

File saved as: 

[1] "Data/Output/CU_34_05_04_meteo_secciones.csv"

#### Copia del fichero a Input

Si el archivo se va a usar en otros notebooks, copiar a la carpeta Input

In [27]:
path_in <- paste0(iPath, file_save)
file.copy(path_out, path_in, overwrite = TRUE)

[1] TRUE

## <font color=' #2874a6 '>Main Conclusions</font>

<font color='steelblue'>List and describe the general conclusions of the analysis carried out​.</font>

### Prerequisites

Para que funcione este código se necesita:

* Las rutas de archivos `Data/Input` y `Data/Output` deben existir (relativas a la ruta del _notebook_)
* El paquete tcltk instalado para seleccionar archivos interactivamente. No se necesita en producción.
* Los paquetes sf, readr, dplyr, tidyr, gstat, stringr y lubridate deben estar instalados.



### Configuration Management

This notebook has been tested with the following versions of R and packages. It cannot be assured that later versions work in the same way:
* R 4.2.2
* tcltk 4.2.2
* sf 1.0.9
* readr 2.1.3
* dplyr 1.0.10
* tidyr 1.3.0
* gstat 2.1.0
* stringr 1.5.0
* lubridate 1.9.1

### Data structures

#### Objeto `data`

- Hay 137392 filas
	* CMUN
	* CDIS
	* CSEC
	* tmed
	* prec
	* velmedia
	* presMax
	* fecha


#### Observaciones generales sobre los datos

* Se incluyen cuatro variables meteorológicas


### Consideraciones para despliegue en piloto

* Ninguna

### Consideraciones para despliegue en producción

* Se deben crear los procesos ETL en producción necesarios para que los datos de entrada estén actualizados

## <font color=' #2874a6 '>Main Actions</font>

#### <font color=' #2874a6 '>Acciones done</font>

<font color='steelblue'>Indicate the actions that have been carried out in this process</font>

- Se han calculado los centroides de las zonas
- Se han interpolado los datos de meteorología a las zonas

#### <font color=' #2874a6 '>Acctions to perform</font>

<font color='steelblue'>Indicate the actions that must be carried out in subsequent processes</font>

- Se debe comprobar por qué se crea una seman 2023-52

## <font color='RED'>CODE TO DEPLOY (PILOT)</font>

A continuación se incluirá el código que deba ser llevado a despliegue para producción, dado que se entiende efectúa operaciones necesarias sobre los datos en la ejecución del prototipo

<font color=' red '><b>Description</b></font>

- No hay nada que desplegar en el piloto, ya que estos datos son estáticos o en todo caso cambian con muy poca frecuencia, altamente improbable durante el proyecto.


<font color=' red '><b>CODE</b></font>

In [28]:
# incluir código